batch/batch/front_end/front_end.py

```diff
@@ -589,11 +626,15 @@ async def _get_full_job_status(app, record):
 @add_metadata_to_request
 async def get_job_log(request, userdata, batch_id):  # pylint: disable=unused-argument
     job_id = int(request.match_info['job_id'])
-    job_log_bytes = await _get_job_log(request.app, batch_id, job_id)
+    job_log_streams = await _get_job_log(request.app, batch_id, job_id)
     job_log_strings: Dict[str, Optional[str]] = {}
-    for container, log in job_log_bytes.items():
+    for container, log in job_log_streams.items():
```

Comment:
I dislike that we don't close these streams. Now that we don't read them in parallel, I'm not sure `_get_job_log` is pulling its weight. Can we refactor to something like `get_job_containers` which gives us the containers and then we can do:
```python3
for container in containers:
    maybe_log = await _get_job_container_log(app, batch_id, job_id, container, record)
    if maybe_log is not None:
        async with maybe_log as log_stream:
           ...
    else:
        ...
```

Reply:
Or: keep `_get_job_log` but make it `_get_truncated_job_logs` which returns 50kB plus a "has more" bit.

Reply:
Tried going the latter route